With more information while sending logs

maskaravivek opened the pull request:
**Description (required)**

Fixes #1961

What changes did you make and why?

Have added the following: 

- App version > will help us identify which version of the app is having the issue
- Username > will help us check if the issue is user specific
- network connection type (wifi/cellular-4g/cellular-3g/cellular-2g/no-internet) > will help us check if the issue is with user's network connection
- the device manufacturer and model > some of the crashes are specific to some device models. Will help us identify if that could be the issue
- Android version > will help us identify if the issue is specific to an android version

**Tests performed (required)**

Tested on `betaDebug`

**Screenshots showing what changed (optional - for UI changes)**

<img width="279" alt="screenshot 2018-12-23 at 2 14 31 pm" src="https://user-images.githubusercontent.com/3069373/50382098-929f2680-06bd-11e9-81fc-14806ab34592.png">

domdomegg commented on app/src/main/java/fr/free/nrw/commons/utils/NetworkUtils.java:
```diff
@@ -22,13 +25,71 @@ public static boolean isInternetConnectionEstablished(@Nullable Context context)
+    static NetworkConnectionType getNetworkType(Context context) {
+        TelephonyManager telephonyManager = (TelephonyManager) context.getApplicationContext().getSystemService(Context.TELEPHONY_SERVICE);
+        NetworkInfo networkInfo = getNetworkInfo(context);
+        NetworkConnectionType networkConnectionType;
```
Could return early instead of nested conditionals?

```suggestion
```

domdomegg commented on app/src/main/java/fr/free/nrw/commons/utils/NetworkUtils.java:
```diff
@@ -22,13 +25,71 @@ public static boolean isInternetConnectionEstablished(@Nullable Context context)
+        } else {
+            int network = networkInfo.getType();
+            if (network == ConnectivityManager.TYPE_WIFI) {
+                networkConnectionType = NetworkConnectionType.WIFI;
```
```suggestion
                return NetworkConnectionType.WIFI;
```

domdomegg commented on app/src/main/java/fr/free/nrw/commons/utils/NetworkUtils.java:
```diff
@@ -22,13 +25,71 @@ public static boolean isInternetConnectionEstablished(@Nullable Context context)
+                    case TelephonyManager.NETWORK_TYPE_EDGE:
+                    case TelephonyManager.NETWORK_TYPE_CDMA:
+                    case TelephonyManager.NETWORK_TYPE_1xRTT:
+                        networkConnectionType = NetworkConnectionType.TWO_G;
```
```suggestion
                        return NetworkConnectionType.TWO_G;
```

domdomegg commented on app/src/main/java/fr/free/nrw/commons/utils/NetworkUtils.java:
```diff
@@ -22,13 +25,71 @@ public static boolean isInternetConnectionEstablished(@Nullable Context context)
+                    case TelephonyManager.NETWORK_TYPE_CDMA:
+                    case TelephonyManager.NETWORK_TYPE_1xRTT:
+                        networkConnectionType = NetworkConnectionType.TWO_G;
+                        break;
```
```suggestion
```

maskaravivek commented:
Thanks for the review @domdomegg. Have addressed the comments. 

Does log file never attach for you? Have you provided storage write permissions to the app? Can you check your phone's storage and see if there's a folder for CommonsApp logs.

misaochan commented:
Looks good to me, thanks @maskaravivek :). @sivaraam has mentioned that the log files don't automatically attach for him, as well. But if everything else works, shall we merge and address that in another issue?

sivaraam commented:
I've created #2265 for the logs not getting attached issue.
